src/tools/rustfmt/src/types.rs

```diff
@@ -930,7 +927,7 @@ fn rewrite_bare_fn(
 
 fn is_generic_bounds_in_order(generic_bounds: &[ast::GenericBound]) -> bool {
     let is_trait = |b: &ast::GenericBound| match b {
-        ast::GenericBound::Outlives(..) => false,
+        ast::GenericBound::Outlives(..) | ast::GenericBound::Use(..) => false,
         ast::GenericBound::Trait(..) => true,
     };
     let is_lifetime = |b: &ast::GenericBound| !is_trait(b);
```

Comment:
This code (and the other rustfmt tweaks) is almost certainly going to handle `use<...>` incorrectly. I'm somewhat concerned that the code is not going to be easy to make correct, and this is likely going to cause more churn if we stabilize `use<...>` without proper formatting.

Reply:
this is kind of tracked in the tracking issue with the `rustfmt` checkbox. Do you think we need to add info to that or is it obvious for any rustfmt implementor?

Reply:
I can add some more FIXMEs to make it obvious